python_modules/dagster/dagster_tests/core_tests/test_object_store.py

```diff
@@ -29,7 +47,30 @@ def test_file_system_object_store():
     finally:
         try:
             shutil.rmtree(object_store.root)
-        except seven.FileNotFoundError:
+        except (seven.FileNotFoundError, OSError):
+            pass
+
+
+def test_file_system_object_store_with_custom_serializer():
+    run_id = str(uuid.uuid4())
+
+    object_store = FileSystemObjectStore(run_id=run_id)
+
+    try:
+        with yield_pipeline_execution_context(
+            PipelineDefinition([]), {}, RunConfig(run_id=run_id)
+        ) as context:
+            object_store.set_object('foo', context, LowercaseString.inst(), ['foo'])
+
+        with open(os.path.join(object_store.root, 'foo'), 'rb') as fd:
+            assert fd.read().decode('utf-8') == 'FOO'
+
+        assert object_store.has_object(context, ['foo'])
+        assert object_store.get_object(context, LowercaseString.inst(), ['foo']) == 'foo'
+    finally:
+        try:
+            shutil.rmtree(object_store.root)
+        except (seven.FileNotFoundError, OSError):
```

Comment:
maybe wrap all this up in a context manager? along with comment about doing nothing in the case is ok. Also what error does OSError represent here?